### PR TITLE
Fixes WB-449: Allow modals to be launched programmatically

### PR DESCRIPTION
**Pull request by kevinbarabash**
The original bug as report was that the focus trap was broken in OneColumnModal.  After some investigation it turned out that modals launched from a menu item were disappearing as soon as the focus changed.  This is because the dropdown menu was closing when this happened which meant that the menu item and the ModalLauncher that was wrapping it were being unmounted.

This PR allows ModalLauncher to be rendered without children and adds an `opened` prop which controls whether the modal is being shown or not.  The reason for going with `opened` and not `open` like some of our other components it to follow the HTML pattern for naming booleans.  I'll be updating the other boolean props to match eventually.

This feature is necessary for some SAT/TestPrep cases where modals are launched on a timer as well as launching modals in response to client-side navigation.

### CODE REVIEW

**Comment by jeresig** [comment]
Deploy preview for *wonder-blocks* ready!

Built with commit 1f90a6055b946331df06d6219e008d23533007bd

https://deploy-preview-326--wonder-blocks.netlify.com

File: packages/wonder-blocks-modal/components/modal-launcher.js [hunk 1]
```diff
@@ -41,6 +41,8 @@ type Props = {|
      * close events.
      */
     onClose?: () => void,
+
+    opened?: boolean,
```
Comment (jeresig) [inline packages/wonder-blocks-modal/components/modal-launcher.js]: Documentation for this, please!

File: packages/wonder-blocks-modal/components/modal-launcher.js [hunk 2]
```diff
@@ -65,6 +67,16 @@ type State = {|
  * the `modal` prop.
  */
 export default class ModalLauncher extends React.Component<Props, State> {
+    static getDerivedStateFromProps(props: Props, state: State) {
+        return {
+            opened: props.opened != null ? props.opened : state.opened,
```
Comment (jeresig) [inline packages/wonder-blocks-modal/components/modal-launcher.js]: I think a slightly better check here would be: `typeof props.opened === "boolean" ? props.opened : state.opened;`

File: package.json [hunk 3]
```diff
@@ -48,6 +48,7 @@
     "concurrently": "^3.6.0",
     "enzyme": "^3.3.0",
     "enzyme-adapter-react-16": "^1.5.0",
+    "enzyme-matchers": "^6.0.4",
```
Comment (jeresig) [inline package.json]: Huh - it looks like this is used but it's confusing that Enzyme's API obscures this...

Comment (kevinbarabash) [reply package.json]: Yeah, I wasn't sure how to set this up.  I couldn't find any docs but it seems to be working after installing it.  I should try uninstalling it to see if the `toExist()` matcher still works.

File: packages/wonder-blocks-modal/components/modal-launcher.md [hunk 4]
```diff
@@ -152,3 +152,79 @@ const oneColumnModal = ({closeModal}) => <OneColumnModal
...
+                )}
+            />
+        </View>;
+    }
+}
+
+<Example />
+```
+
+**Warning:** Do not wrap items within an dropdown in a `ModalLauncher`.  It causes the
```
[the remark this answers is not in the document]
Comment (kevinbarabash) [reply packages/wonder-blocks-modal/components/modal-launcher.md]: I was going to limit what the `children` function can return to only be things that work well with that pattern like `Button`.

File: packages/wonder-blocks-modal/components/modal-launcher.md [hunk 5]
```diff
@@ -152,3 +152,79 @@ const oneColumnModal = ({closeModal}) => <OneColumnModal
...
+modal to disappear as soon as the focus changes.  The reason is that the change in
+focus results in the item that in the dropdown that was clicked to be blur which
+closes the dropdown.  This results in all of its children to unmount including the
+ModalLauncher which was wrapping the menu item.
+
+**TODO:** use flow to prevent this.
```
[the remark this answers is not in the document]
Comment (kevinbarabash) [reply packages/wonder-blocks-modal/components/modal-launcher.md]: I've created 
https://khanacademy.atlassian.net/browse/WB-462 to track this.

File: packages/wonder-blocks-modal/components/modal-launcher.md (hunk 4 above)
[the remark this answers is not in the document]
Comment (kevinbarabash) [reply packages/wonder-blocks-modal/components/modal-launcher.md]: I'll do this in a separate PR.

**Pull request merged**